### PR TITLE
[openstack] Allow custom cloud names

### LINKED ISSUE #303: [openstack] should not use hard code 'openstack' cloud
/bug

https://github.com/openshift/cluster-image-registry-operator/blob/52dfda7/pkg/clusterconfig/clusterconfig.go#L194

made assumption that the 'openstack' cloud is in cloud defintion but that's not always true and lead to https://github.com/openshift/installer/pull/1858 having issue in pass CI test

### PR DESCRIPTION
**Pull request by adambkaplan**
Fixes #303

### CODE REVIEW

File: pkg/storage/swift/swift.go [hunk 1]
```diff
@@ -71,7 +71,23 @@ func GetConfig(listers *regopclient.Listers) (*Swift, error) {
...
+				if !errors.IsNotFound(err) {
+					return nil, fmt.Errorf("failed to get cluster infrastructure info: %v", err)
+				}
+			}
+			if cloudInfra != nil &&
+				cloudInfra.Status.PlatformStatus != nil &&
+				cloudInfra.Status.PlatformStatus.OpenStack != nil {
+				cloudName = cloudInfra.Status.PlatformStatus.OpenStack.CloudName
+			}
+			if len(cloudName) == 0 {
```
Comment (coreydaley) [inline pkg/storage/swift/swift.go]: Are you also trying to capture if cloudInfra.Status.PlatformStatus.OpenStack.CloudName is blank here?
Just curious.

File: pkg/storage/swift/swift_test.go [hunk 2]
```diff
@@ -144,6 +147,25 @@ func handleAuthentication(t *testing.T, endpointType string) {
 	})
 }
 
+func fakeInfrastructureLister(cloudName string) configlisters.InfrastructureLister {
```
Comment (coreydaley) [inline pkg/storage/swift/swift_test.go]: Can this go here?
https://github.com/openshift/cluster-image-registry-operator/tree/master/test/framework/mock/listers

Also, should some of these tests in this file really be in e2e?

File: pkg/storage/swift/swift.go (hunk 1 above)
Comment (adambkaplan) [reply pkg/storage/swift/swift.go]: Yes, so we can unblock the install team. If we don't have the infra object, or the platform status doesn't have the cloud name filled yet, fall back to `openstack`.

**Comment by adambkaplan** [comment]
/hold

Until issues related to openshift controller manager are resolved

**Pull request merged**